src/frontend/components/DashboardThumbnail/index.tsx

```diff
@@ -50,21 +50,27 @@ export const DashboardThumbnail = ({ video }: DashboardThumbnailProps) => {
     thumbnail: state.getThumbnail(),
   }));
 
+  const thumbnailState = thumbnail ? thumbnail.upload_state : uploadState.READY;
+
+  const cleanup = () => {
+    if (pollInterval > -1) {
+      window.clearInterval(pollInterval);
+    }
+  };
+
   useEffect(() => {
     if (
       thumbnail &&
-      thumbnail.upload_state === uploadState.PROCESSING &&
+      [uploadState.PROCESSING, uploadState.PENDING].includes(
+        thumbnail.upload_state,
+      ) &&
       pollInterval === -1
     ) {
       setPollInterval(window.setInterval(() => pollThumbnail(), 1000 * 5));
```

Comment:
I think we could use a `ref` to store the `setInterval` id.
In this way, we save one render call when `pollInterval` is updated